### PR TITLE
Prepare whawty.auth for auxiliary data

### PR DESCRIPTION
**Pull request by nicoonoclaste**
- [x] Make `userhash` only read and parse the first line of user files.
- [x] Make `userhash` preserve the auxiliary data when updating password hashes.
- [x] Make base updates atomic (required, since several agents may update the auxiliary data)
  - Document (in `doc/SCHEMA`) a path for temporary files and a mechanism for atomic updates.
  - Add a `getTempFile` helper in `store`
  - Update `writeUserHash` accordingly, replace its `flag` parameter by `mayCreate`

### CODE REVIEW

**Comment by nicoonoclaste** [comment]
This still need tests:
- [ ] Auxiliary data is indeed preserved.
- [ ] Auxiliary data (or lack thereof) doesn't affect the parsed `userhash`

**Comment by nicoonoclaste** [comment]
@equinox0815 OK for merge & moving the comment about missing tests to an issue?

File: store/store.go [hunk 1]
```diff
@@ -113,6 +114,23 @@ func openDir(path string) (*os.File, error) {
 	return dir, nil
 }
 
+// getTempFile provides a new, empty file in the base's .tmp directory,
+//  suitable for atomic file updates (by create/write/rename)
+func (dir *Dir) getTempFile() (tmp *os.File, err error) {
+	tmpDir := path.Join(dir.BaseDir, ".tmp")
```
Comment (equinox0815) [inline store/store.go]: please make the ".tmp" a const like userExt or adminExt (see top of file)

File: store/store.go [hunk 2]
```diff
@@ -192,6 +210,11 @@ func listSupportedUsers(dir *os.File, list UserList) error {
 		}
 
 		for _, name := range names {
+			// Skip the '.tmp' directory
+			if name == ".tmp" {
```
Comment (equinox0815) [inline store/store.go]: use const from above

File: store/store.go (hunk 1 above)
Comment (nicoonoclaste) [reply store/store.go]: Good point

**Pull request merged**